org.lflang/src/org/lflang/validation/LFValidator.java

```diff
@@ -1313,6 +1308,79 @@ public void checkModeInstanceNamespace(Reactor reactor) {
         }
     }
 
+    @Check(CheckType.FAST)
+    public void checkMissingStateResetInMode(Reactor reactor) {
```

Comment:
We should probably add some unit tests for this...

Reply:
Added unit tests for all mode related validation rules.